lib/admin/db_drift.dart

```diff
@@ -201,18 +204,65 @@ class MyDatabase extends _$MyDatabase {
     String description,
     String parent,
   ) async {
+    final parentAsset = await (select(assets)
+          ..where((t) => t.assetnum.equals(parent))
+          ..where((t) => t.siteid.equals(siteid)))
+        .getSingle();
+    final hierarchy = parentAsset.hierarchy! + "," + assetnum;
+
     final row = await into(assets).insertReturning(AssetsCompanion.insert(
       description: description,
+      hierarchy: Value(hierarchy),
       assetnum: assetnum,
       siteid: siteid,
       parent: Value(parent),
       priority: 0,
       status: 'Planning',
       changedate: 'N/A',
+      newAsset: Value(1),
     ));
     return row.id;
   }
 
+  Future<int> deleteAsset(String assetNum, String siteId) async {
+    final row = await (delete(assets)
+          ..where((t) => t.assetnum.equals(assetNum))
+          ..where((t) => t.siteid.equals(siteId)))
+        .goAndReturn();
+    return row.first.id;
+  }
+
+
+  ///0 = existing asset
+  ///
+  ///1 = new asset
+  ///
+  ///-1 = asset that failed upload (e.g. location and asset were uploaded, but job plan failed)
+  Future<int> setAssetStatus(
+      String assetNum, String siteId, int assetStatus) async {
+    var existingAsset = await getAsset(siteId, assetNum);
+    var newAsset = Asset(
+        assetnum: existingAsset.assetnum,
+        changedate: existingAsset.changedate,
```

Comment:
seems extra, see first example
https://drift.simonbinder.eu/docs/getting-started/writing_queries/#updates-and-deletes